File: cases/taint/kotlin/direct-negative/DirectFlow.kt
package dataflowbench

object DirectFlow {
    fun dfb_source(): String { // DFB-SOURCE: direct-input
        return "tainted"
    }

    fun dfb_sink(value: String) {} // DFB-SINK: direct-sink

    fun run() {
        dfb_source()
        dfb_sink("clean")
    }
}

File: cases/taint/csharp/direct-negative/DirectFlow.cs
namespace DataFlowBench;

static class DirectFlow
{
    static string dfb_source() // DFB-SOURCE: direct-input
    {
        return "tainted";
    }

    static void dfb_sink(string value) { } // DFB-SINK: direct-sink

    static void Run()
    {
        dfb_source();
        dfb_sink("clean");
    }
}
